oscar_odin/mappings/context.py

```diff
@@ -210,16 +223,39 @@ def bulk_update_or_create_one_to_many(self):
 
         for relation, instances in instances_to_create.items():
             validated_instances_to_create = self.validate_instances(instances)
+            instance_identifiers[relation] = dict()
```

Comment:
hm these are quite a lot of for loops here.

Is there no better solution? is it not possible to add some functions to seperate_instances_to_create_and_update and add delete to there?